cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -1242,6 +1242,21 @@ TEST_F(ScalarTemporalTest, TestTemporalSubtractTimeAndDuration) {
   }
 }
 
+TEST_F(ScalarTemporalTest, TestTemporalSubtractDuration) {
+  for (auto op : {"subtract", "subtract_checked"}) {
+    for (auto u : TimeUnit::values()) {
+      auto unit = duration(u);
+      CheckScalarBinary(op, ArrayFromJSON(unit, times_s2), ArrayFromJSON(unit, times_s),
+                        ArrayFromJSON(unit, seconds_between_time));
+    }
+
+    auto seconds_3 = ArrayFromJSON(duration(TimeUnit::SECOND), R"([3, null])");
+    auto milliseconds_2k = ArrayFromJSON(duration(TimeUnit::MILLI), R"([2000, null])");
+    auto milliseconds_1k = ArrayFromJSON(duration(TimeUnit::MILLI), R"([1000, null])");
+    CheckScalarBinary(op, seconds_3, milliseconds_2k, milliseconds_1k);
+  }
+}
```

Comment:
Can you add tests for overflowing inputs?

Reply:
Added. However it seems as it duration is not treated as `int64_t` [here](https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/int_util_internal.h#L59) and overflow is not detected?

Reply:
Never mind. I was using the wrong operator.